presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1254,10 +1255,24 @@ private boolean isExpensiveUnionDistinct(SetOperation setOperation, Type[] outpu
                                             type instanceof RowType);
         }
 
+        @Override
+        protected Scope visitUnion(Union node, Optional<Scope> scope)
+        {
+            if (!node.isDistinct().isPresent()) {
+                warningCollector.add(new PrestoWarning(PARSER_WARNING,
```

Comment:
@kaikalur Should this be a parser warning or a performance warning? I'm debating on this...

Reply:
Should be a performance warning given by the parser :)

Reply:
Changed to performance warning